src/components/Announcements/Announcements.tsx

```diff
@@ -0,0 +1,184 @@
+      text={process.env.ALERT_MESSAGE}
+      localStorageName="ALERT_MESSAGE"
+    />
+  ) : null;
+};
+
+const Announcement: React.FC = () => {
+  const { push } = useRouter();
+  const accountListId = useAccountListId();
+  const [showAppealModal, setShowAppealModal] = useState(false);
+  const [hideAnnouncement, setHideAnnouncement] = useState(false);
+  const { contactStatuses } = useContactPartnershipStatuses();
+
+  const { data } = useAnnouncementsQuery();
+  const [acknowledgeAnnouncement] = useAcknowledgeAnnouncementMutation();
+
+  // TODO replace this with the ability to only limit announcements to one item.
+  const announcement = useMemo(() => {
+    if (data?.announcements.nodes.length) {
+      return data.announcements.nodes[0];
+    }
+    return null;
+  }, [data]);
+
+  // The `useEffect` hook ensures that the styles are only loaded when necessary
+  // and are cleaned up when the component is unmounted or the announcement changes.
+  useEffect(() => {
+    if (!announcement) {
+      return;
+    }
+    const hasBeenLoaded = document.querySelector(
+      'link[id="fontAwesomeStyles"]',
+    );
+    if (!hasBeenLoaded) {
+      const fontAwesomeStyles = document.createElement('link');
+      fontAwesomeStyles.rel = 'stylesheet';
+      fontAwesomeStyles.id = 'fontAwesomeStyles';
+      fontAwesomeStyles.href =
+        'https://use.fontawesome.com/releases/v5.14.0/css/all.css';
+      document.head.appendChild(fontAwesomeStyles);
+    }
+
+    return () => {
+      const fontAwesomeStyles = document.querySelector(
+        'link[id="fontAwesomeStyles"]',
+      );
+      if (fontAwesomeStyles) {
+        document.head.removeChild(fontAwesomeStyles);
+      }
+    };
+  }, [announcement]);
+
+  const handleAcknowledge = async (
+    announcementId: string,
+    actionId?: string,
+  ) => {
+    if (!announcementId) {
+      return;
+    }
+
+    await acknowledgeAnnouncement({
+      variables: { input: { announcementId, actionId } },
+    });
+  };
+
+  const handlePerformAction = useCallback(
+    (action?: ActionFragment) => {
+      if (action) {
+        switch (action.action) {
+          case ActionEnum.Go:
+            if (action.args) {
+              // TODO - Migrate the data on production, to convert to "Go" args to be to the new MPDX URL format
+              push(`/accountLists/${accountListId}/${action.args}`);
+            }
+            break;
+          case ActionEnum.AppealCreate:
+            setShowAppealModal(true);
+            break;
+          case ActionEnum.Track:
+            const userRatingMatch = action.args?.match(
+              /^aa-(user-rating)-(\d)$/,
+            );
+            if (userRatingMatch) {
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              const [_adobeEventName, event, rating] = userRatingMatch;
+              dispatch(event, { rating });
+            } else {
+              dispatch(action?.args ?? 'undefined event');
+            }
```

Comment:
I know this is the last case, but it's probably good to have a `break;` here anyway.

Reply:
I'll add it in so it makes it easier to read.